activerecord/lib/active_record/persistence.rb

```diff
@@ -561,6 +565,10 @@ def delete(id_or_array)
         delete_by(primary_key => id_or_array)
       end
 
+      def reload_attributes_on_save(*attributes) # :nodoc:
```

Comment:
Not sure what the workflow preferences are for the Rails team but my thinking is if this PR looks acceptable we can merge now but keep this method undocumented until the two follow-up PRs are complete (RETURNING on update and SELECT for mysql).   Once those are done we can add documentation to make it a public method.